src/rt_tokio_wasi/mod.rs

```diff
@@ -0,0 +1,71 @@
+use std::future::Future;
+use std::io;
+use std::marker::PhantomData;
+
+use once_cell::sync::Lazy;
+
+static RUNTIME: Lazy<tokio::runtime::Runtime> = Lazy::new(|| {
+    tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .unwrap()
+});
```

Comment:
If a platform has no multi-threading support, i.e.: a thread runs on the main thread, the runtime should already be running via `#[tokio::main]`. Creating a different runtime assumed to be run in the same thread is not going to work as calling `block_on` effectively blocks the outer runtime.

The correct way to run a runtime in the main thread is to call `tokio::task::LocalSet::run_until()` in `main()`.

See: https://docs.rs/tokio/latest/tokio/task/struct.LocalSet.html#method.run_until